### PR TITLE
feat: Add support for local and remote ipv4 cidr

### PR DESCRIPTION
**Pull request by juho9000**
## Description
Add support for local_ipv4_network_cidr remote_ipv4_network_cidr parameters of aws_vpn_connection resource

## Motivation and Context
https://github.com/terraform-aws-modules/terraform-aws-vpn-gateway/issues/44

## Breaking Changes
Not exactly a breaking change, but maybe good to know; Running pre-commit updates the provider in docs to the one from Terraform lockfile, see: https://github.com/terraform-docs/terraform-docs/issues/517 

## How Has This Been Tested?
- [x] I have tested and validated these changes using one or more of the provided `examples/*` projects

### CODE REVIEW

**Comment by lgallard** [comment]
@antonbabenko any update on this PR?

File: variables.tf [hunk 1]
```diff
@@ -247,3 +247,13 @@ variable "tunnel2_ike_versions" {
   type        = list(string)
   default     = null
 }
+variable "local_ipv4_network_cidr" {
+  description = "(Optional) The IPv4 CIDR on the customer gateway (on-premises) side of the VPN connection."
+  type        = string
+  default     = null
```
Comment (lgallard) [inline variables.tf:253]: I would suggest to use [the default value for the local IPv4 CIDR provided by AWS](https://docs.aws.amazon.com/vpn/latest/s2svpn/VPNTunnels.html#:~:text=Local%20IPv4%20Network%20CIDR) here, which is `0.0.0.0/0`

File: variables.tf [hunk 2]
```diff
@@ -247,3 +247,13 @@ variable "tunnel2_ike_versions" {
...
+}
+variable "remote_ipv4_network_cidr" {
+  description = "(Optional) The IPv4 CIDR on the AWS side of the VPN connection."
+  type        = string
+  default     = null
```
Comment (lgallard) [inline variables.tf:258]: Same here, I would suggest to use [the default value for the remote IPv4 CIDR provided by AWS](https://docs.aws.amazon.com/vpn/latest/s2svpn/VPNTunnels.html#:~:text=Remote%20IPv4%20Network%20CIDR) here, which is `0.0.0.0/0`

**Comment by khos2ow** [comment]
FYI we've just released terraform-docs [`v0.15.0`](https://github.com/terraform-docs/terraform-docs/releases/tag/v0.15.0) which allows for ignoring reading `.terraform.lock.hcl`. This PR can partially be updated to incorporate the new change, or just leave it as is.

**Comment by juho9000** [comment]
Thanks for reviewing, I've changed the default values to used AWS defaults as suggested.

**Comment by antonbabenko** [comment]
Thanks @juho9000 !

@khos2ow [Next week](https://www.youtube.com/watch?v=D42CCLSA-qU) we will probably discuss and then implement a better solution for the `terraform-docs`. Thanks for your feedback!

[v2.11.0](https://github.com/terraform-aws-modules/terraform-aws-vpn-gateway/releases/v2.11.0) has been just released.

**Pull request merged**